src/common/config/state/init.js

```diff
@@ -55,14 +55,14 @@ const model = () => {
         return hiddenState.block_selector
       },
       set block_selector (val) {
-        hiddenState.block_selector += `,${val}`
+        if (val) hiddenState.block_selector += `,${val}`
```

Comment:
Need to be a little more diligent here I think.

```
const queryCheck = (s) => document.createDocumentFragment().querySelector(s)
const isSelectorValid = (selector) => {
  try { queryCheck(selector) } catch { return false }
  return true
}
...
const hiddenState = {
  block_selector: new Set(['[data-nr-block]']),
  mask_input_options: { password: true }
}
...
get block_selector () {
  return Array.from(hiddenState.block_selector)
}
set block_selector (val) {
  if (typeof val !== 'string') return
  val.split(',').forEach(selector => {
    if (isSelectorValid(selector)) hiddenState.block_selector.add(selector)
    else warn(`An invalid session_replay.block_selector was provided: ${selectpr}`)
  })
}
```

Reply:
normally i'm not a big proponent of agent-side validation but this makes sense since it is critical to blocking sensitive data :+1:

Reply:
probably should apply this validSelector check to other selectors as well, like mask-text selector

Reply:
The other issue, or reason to do this, would be that an error is thrown to the console if an invalid selector is provided.